daemon/cluster/noderunner.go

```diff
@@ -115,24 +115,20 @@ func (n *nodeRunner) start(conf nodeStartConfig) error {
 		joinAddr = conf.RemoteAddr
 	}
 
-	var defaultAddrPool []*net.IPNet
-	for _, address := range conf.DefaultAddressPool {
-		if _, b, err := net.ParseCIDR(address); err == nil {
-			defaultAddrPool = append(defaultAddrPool, b)
-		}
-	}
 	// Hostname is not set here. Instead, it is obtained from
 	// the node description that is reported periodically
 	swarmnodeConfig := swarmnode.Config{
 		ForceNewCluster:    conf.forceNewCluster,
 		ListenControlAPI:   control,
 		ListenRemoteAPI:    conf.ListenAddr,
 		AdvertiseRemoteAPI: conf.AdvertiseAddr,
-		DefaultAddrPool:    defaultAddrPool,
-		SubnetSize:         int(conf.SubnetSize),
-		JoinAddr:           joinAddr,
-		StateDir:           n.cluster.root,
-		JoinToken:          conf.joinToken,
+		NetworkConfig: &swarmallocator.NetworkConfig{
```

Comment:
Also wondering if we should convert to `[]*ipamutils.NetworkToSplit` earlier, instead of passing IP-ranges and Subnetsize separate here

Reply:
In this way, again we will be extracting info from NetworkToSplit to string slice to store info into raft.  That was one main comment from Anshul where he wants to avoid converting information back and forth. In this way we know only swarmkit needs to know what format libnetwork expects. we dont need to push the information at Moby level.